### PR TITLE
Feat: Global context, Product filters

### PR DESCRIPTION
**Pull request by pritam-kr**
Note: Till now I did not update to view all things happing on the console. 

- [Feat: Created global context, Sort by Price, Filter by Category ](https://github.com/pritam-kr/Pearl/commit/db82335a9859df187daf2fa07d6d1c5bdedc078a)
- Fix: product database
- [global context and filters](https://github.com/pritam-kr/Pearl/commit/db82335a9859df187daf2fa07d6d1c5bdedc078a) 

Screenshot (2)

Note: Till now I did not update to view all things happing on the console.

### CODE REVIEW

File: src/Context/GlobalContext/StateContext.jsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import { createContext, useContext, useReducer, useEffect, useState } from "react"
+import axios from "axios"
+import { uniqueCategory } from "../../utils/uniqueCategory"
+import { sortByPrice } from "../../utils/sortByPrice"
+import { stateReducerFun } from "../GlobalContext/StateReducer"
+
+const StateContext = createContext()
+
```
Comment (poornima-alugubelly) [inline src/Context/GlobalContext/StateContext.jsx:8]: maybe u can name this globalstatecontext for more specificity

File: src/Context/GlobalContext/StateContext.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    const getFilterByCategory = filterByCategory(getSortByPrice, categoryName)
+    console.log(getFilterByCategory)
+
+    useEffect(() => {
+        (async () => {
+            try {
+                const { data: { products } } = await axios.get('/api/products')
+                dispatch({ type: "ON_SUCCESS", payload: products, setLoading: setLoading })
+            }
```
Comment (poornima-alugubelly) [inline src/Context/GlobalContext/StateContext.jsx:54]: maybe you can also verify whether data is coming or not and then call dispatch?

**Review by poornima-alugubelly — APPROVED** [review]

File: src/Context/GlobalContext/StateContext.jsx (hunk 1 above)
Comment (pritam-kr) [reply src/Context/GlobalContext/StateContext.jsx:8]: Okay

File: src/Context/GlobalContext/StateContext.jsx (hunk 2 above)
Comment (pritam-kr) [reply src/Context/GlobalContext/StateContext.jsx:54]: Yes, I'll add it later.

File: src/App.css [hunk 3]
```diff
@@ -81,7 +81,7 @@
   }
   
   label {
-    font-family: "Zen Kaku Gothic Antique", sans-serif;
+    font-family: "Zen Kaku Gothic Antique", sans-serif ;
```
Comment (horsemaker) [inline src/App.css:84]: Is this an intentional commit?

File: src/Context/GlobalContext/StateContext.jsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+const initialState = {
+    products: [],
+    filters: {
+        sortBy: '',
+        priceRange: 20,
+        categoryName: [],
+        rating: ''
+    },
+    wishlist: [],
+}
```
Comment (msdeshmukh009) [inline src/Context/GlobalContext/StateContext.jsx:19]: Just a suggestion but you can have more than one logically separated context as well. 
For example, You can create a different context for filters and wishlist.

**Review by msdeshmukh009 — APPROVED** [review]
That video on the home page is looking really nice. Grea work!!

**Review by horsemaker — APPROVED** [review]

File: src/App.css (hunk 3 above)
Comment (pritam-kr) [reply src/App.css:84]: It is a mistake, I'll fix it.

**Pull request merged**